--- fail2ban.orig	2008-12-31 21:31:12.000000000 +1000
+++ fail2ban	2008-06-24 14:20:53.000000000 +1000
@@ -44,6 +20,7 @@
 my $DebugCounter = 0;
 my $ReInitializations = 0;
 my @IptablesErrors = ();
+my @ActionErrors = ();
 my $NotValidIP = 0;		# reported invalid IPs number
 my @OtherList = ();
 
@@ -63,6 +40,7 @@
     if ( ($ThisLine =~ /..,... DEBUG: /) or
 	 ($ThisLine =~ /..,... \S*\s*: DEBUG /) or # syntax of 0.7.? fail2ban
 	 ($ThisLine =~ /..,... INFO: (Fail2Ban v.* is running|Exiting|Enabled sections:)/) or
+	 ($ThisLine =~ /INFO.*Log rotation detected for/) or
 	 ($ThisLine =~ /..,... \S+\s*: INFO\s+(Set |Socket|Exiting|Gamin|Created|Added|Using)/) or # syntax of 0.7.? fail2ban
 	 ($ThisLine =~ /..,... WARNING: Verbose level is /) or
 	 ($ThisLine =~ /..,... WARNING: Restoring firewall rules/)
@@ -84,10 +62,14 @@
 	push @{$ServicesBans{$Service}{$Host}{'Failures'}}, $NumFailures;
     } elsif ( my ($Service,$Host) = ($ThisLine =~ m/ ERROR:\s(.*):\s(\S+)\salready in ban list/)) {
 	$ServicesBans{$Service}{$Host}{'AlreadyInTheList'}++;
+    } elsif ( my ($Service,$Host) = ($ThisLine =~ m/WARNING\s*\[(.*)\]\s*(\S+)\s*already banned/)) {
+	$ServicesBans{$Service}{$Host}{'AlreadyInTheList'}++;
     } elsif ( my ($Service,$Host) = ($ThisLine =~ m/ WARNING:\s(.*):\sReBan (\S+)/)) {
 	$ServicesBans{$Service}{$Host}{'ReBan'}++;
     } elsif ($ThisLine =~ / ERROR:?\s*(Execution of command )?\'?iptables/) {
 	push @IptablesErrors, "$ThisLine\n";
+    } elsif ($ThisLine =~ /ERROR.*returned \d+$/) {
+	push @ActionErrors, "$ThisLine\n";
     } elsif (($ThisLine =~ /..,... WARNING: \#\S+ reinitialization of firewalls/) or
 	     ($ThisLine =~ / ERROR\s*Invariant check failed. Trying to restore a sane environment/)) {
 	$ReInitializations++;
@@ -143,6 +125,13 @@
 	    print @IptablesErrors ;
 	}
     }
+    if ($#ActionErrors > 0) {
+	printf("\n%d error(s) returned from actions", $#ActionErrors);
+	if ($Detail > 5) {
+	    print ":\n";
+	    print @ActionErrors ;
+	}
+    }
     if ($ReInitializations > 0) {
 	printf("\n%d fail2ban rules reinitialization(s)", $ReInitializations);
     }
